src/test/arith_uint256_tests.cpp

```diff
@@ -278,6 +281,12 @@ BOOST_AUTO_TEST_CASE( comparison ) // <= >= < >
         BOOST_CHECK( R1L <= TmpL ); BOOST_CHECK( (R1L == TmpL) != (R1L < TmpL)); BOOST_CHECK( (TmpL == R1L) || !( R1L >= TmpL));
         BOOST_CHECK(! (TmpL < R1L)); BOOST_CHECK(! (R1L > TmpL));
     }
+
+    BOOST_CHECK_LT(ZeroL,
+                   OneL);
+    // Verify hex number representation has the most significant bytes first.
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/30526#discussion_r1699168468

Still think it would be better to say digits instead of bytes here because this is referring to a number string, not a byte array.

re: https://github.com/bitcoin/bitcoin/pull/30526#discussion_r1700191168

> > When there's a hex encoding, "byte" could refer either to a byte of the hex string or to a byte of the data it represents, and I think you mean the former (because the digits in the two strings are what's reversed, not the bytes).
> 
> I mean that `FF` represents a byte will all bits set, so I guess the latter. The individual hex digits are NOT reversed, `0F` is not interpreted as `F0`, it is the order of the _bytes_ that is reversed, `12EF` => `EF12`.

I don't understand why you are thinking about bytes at all here when this is a number string. When I am saying the digits are reversed I am saying "12345" is becoming "54321". I am not saying "12345" is becoming "103254". In your test you making sure the number "0000000000000000000000000000000000000000000000000000000000000001" is less than the number "1000000000000000000000000000000000000000000000000000000000000000". The digits in these numbers are reversed, not the bytes. The second number is straightforwardly higher than the first because the highest digit is bigger (1 > 0). It is true you can also say that the highest byte is bigger (10 > 0), but it's an odd thing to say because this is just a number and there should be no need to think about bytes at all to make sure a numeric comparison is working correctly.